sentry_sdk/integrations/asgi.py

```diff
@@ -185,66 +183,47 @@ async def _run_app(self, scope, receive, send, asgi_version):
                         scope,
                     )
 
-                    if ty in ("http", "websocket"):
-                        transaction = continue_trace(
-                            _get_headers(scope),
-                            op="{}.server".format(ty),
+                    with sentry_sdk.continue_trace(_get_headers(scope)):
+                        with sentry_sdk.start_transaction(
+                            op=(
+                                OP.WEBSOCKET_SERVER
+                                if ty == "websocket"
+                                else OP.HTTP_SERVER
+                            ),
                             name=transaction_name,
                             source=transaction_source,
                             origin=self.span_origin,
-                        )
-                        logger.debug(
-                            "[ASGI] Created transaction (continuing trace): %s",
-                            transaction,
-                        )
-                    else:
```

Comment:
We're removing this `else` logic for non-`http` and non-`websocket` scope types, was this intentional? I'm not sure why it was there in the first place (maybe future proofing for new asgi scope types?), but it seems deliberate

Reply:
Ok this is why: https://github.com/getsentry/sentry-python/commit/44c43a8b7446b74f4c2bdf24e2938247fd17f5ac Apparently fetching headers might be problematic for other scope types. So we should probably keep this

Reply:
thanks for digging, made the accessor safe to avoid the if 
https://github.com/getsentry/sentry-python/pull/3486/commits/02efbe677820880706e6a3213e3c4a5a78aa490f